### PR TITLE
pvc multiple clone performance

### PR DESCRIPTION
**Pull request by ypersky1980**

### CODE REVIEW

File: tests/e2e/performance/test_pvc_clone_performance.py [hunk 1]
```diff
@@ -124,10 +140,10 @@ def test_clone_create_delete_performance(self, interface_type, pvc_size, file_si
             clone_yaml = constants.CSI_CEPHFS_PVC_CLONE_YAML
         file_size_mb = convert_device_size(file_size, "MB")
 
-        # creating single clone ( or many one by one if max_mum_of_clones > 1)
-        logger.info(f"Start creating {max_num_of_clones} clones on {interface_type} PVC of size {pvc_size} GB.")
+        # creating clone(s)
+        # logger.info(f"Start creating {clones_num} clones on {interface_type} PVC of size {pvc_size} GB.")
```
Comment (ebenahar) [inline tests/e2e/performance/test_pvc_clone_performance.py:144]: should this be commented out?

File: tests/e2e/performance/test_pvc_clone_performance.py [hunk 2]
```diff
@@ -79,14 +50,60 @@ def base_setup(self, interface_type, pvc_size, pvc_factory, pod_factory):
...
+            ),
+            pytest.param(
+                *[constants.CEPHFILESYSTEM, '50', '30Gi'], marks=pytest.mark.polarion_id('OCS-2359')
+            ),
+            pytest.param(
+                *[constants.CEPHFILESYSTEM, '100', '60Gi'], marks=pytest.mark.polarion_id('OCS-2360')
+            )
+        ]
+    )
+    @pytest.mark.usefixtures(base_setup_single_clone.__name__)
```
Comment (ebenahar) [inline tests/e2e/performance/test_pvc_clone_performance.py:83]: suggest following the convention of passing the fixture name to the test function:

```python
def test_single_clone_create_delete_performance(self, interface_type, pvc_size, file_size, teardown_factory, base_setup_single_clone)
```

**Review by AviadP — COMMENTED** [review]
Please Provide meaningful description or link to related issue in PR description, also, plz remove your name from PR title

File: tests/e2e/performance/test_pvc_clone_performance.py [hunk 3]
```diff
@@ -188,4 +204,87 @@ def test_clone_create_delete_performance(self, interface_type, pvc_size, file_si
...
+    def base_setup_multiple_clones(self, interface_type, pvc_factory, pod_factory):
+        """
+        create resources for the test_multiple_clone_create_delete_performance
+        and calculates storage size, pvc size for the pvc and clones to be created
+        exits the test if there is not enough storage size for pvc and its multiple clones
+        Args:
+            interface_type(str): The type of the interface
+                (e.g. CephBlockPool, CephFileSystem)
+            pvc_size: Size of the created PVC
+            pvc_factory: A fixture to create new pvc
```
Comment (AviadP) [inline tests/e2e/performance/test_pvc_clone_performance.py:218]: It is not needed to add fixtures description when they used as function arguments

File: tests/e2e/performance/test_pvc_clone_performance.py [hunk 4]
```diff
@@ -97,7 +114,7 @@ def test_clone_create_delete_performance(self, interface_type, pvc_size, file_si
         self.pod_obj.fillup_fs(size=file_size_for_io, fio_filename=file_name)
 
         # Wait for fio to finish
-        fio_result = self.pod_obj.get_fio_results()
+        fio_result = self.pod_obj.get_fio_results(timeout=18000)
```
Comment (keesturam) [inline tests/e2e/performance/test_pvc_clone_performance.py:117]: What is the typical time to complete fio job? 18000 for timeout seems like a huge value.

File: tests/e2e/performance/test_pvc_clone_performance.py [hunk 5]
```diff
@@ -124,10 +140,10 @@ def test_clone_create_delete_performance(self, interface_type, pvc_size, file_si
...
 
-        # creating single clone ( or many one by one if max_mum_of_clones > 1)
-        logger.info(f"Start creating {max_num_of_clones} clones on {interface_type} PVC of size {pvc_size} GB.")
+        # creating clone(s)
+        # logger.info(f"Start creating {clones_num} clones on {interface_type} PVC of size {pvc_size} GB.")
 
-        for i in range(max_num_of_clones):
+        for i in range(clones_num):
             logger.info(f'Start creation of clone number {i + 1}.')
             cloned_pvc_obj = pvc.create_pvc_clone(sc_name, parent_pvc, clone_yaml, storage_size=pvc_size + "Gi")
```
Comment (keesturam) [inline tests/e2e/performance/test_pvc_clone_performance.py:148]: creating objects for each clone is memory intensive like we've seen in the past. we may need to enhance this later.

**Review by keesturam — COMMENTED** [review]
Left a few comments.

**Pull request closed without merging**